sys/kern/kern_physio.c

```diff
@@ -154,13 +154,13 @@ physio(struct cdev *dev, struct uio *uio, int ioflag)
 			 * larger than MAXPHYS - PAGE_SIZE must be
 			 * page aligned or it will be fragmented.
 			 */
-			poff = (vm_offset_t)bp->bio_data & PAGE_MASK;
+			poff = (__cheri_addr vm_offset_t)base & PAGE_MASK;
 			if (pbuf && bp->bio_length + poff > pbuf->b_kvasize) {
 				if (dev->si_flags & SI_NOSPLIT) {
 					uprintf("%s: request ptr %p is not "
 					    "on a page boundary; cannot split "
 					    "request\n", devtoname(dev),
-					    bp->bio_data);
+					    base);
```

Comment:
Wrong format string for hybrid kernels (probably easiest to just cast it rather than deal with the mess of %p/%lp and indirect passing)